### ISSUE TITLE
JOSS Review

### ISSUE THREAD

**Issue by Fil** [flekschas/regl-scatterplot#106]
Ref. https://github.com/openjournals/joss-reviews/issues/5275

The only issue I found is:
* community guidelines are missing

_paper_

* It is technically incorrect to say that D3 relies on SVG (it can do canvas too, for instance).
* typos:
  * s/spacial/spatial/
  * "study, end test" s/end/and/
 

_example_

the [observable example](https://observablehq.com/@flekschas/regl-scatterplot) could be enhanced by adding interaction on selection; for example it could be as simple as adding 2 cells and changing 1:

~~~diff
+ mutable selection = null

+ scatterplot.subscribe("select", ({points}) => mutable selection = points)

- createScatterplot = require('regl-scatterplot@0.7.6/dist/regl-scatterplot.min.js')
+ createScatterplot = require('regl-scatterplot@1')
~~~

(Or it could be rewritten in a more "observable" way by using [viewof and data flow](https://observablehq.com/@fil/hello-regl-scatterplot); see also https://observablehq.com/@radames/hello-regl-scatterplot for earlier work.)

**Comment by fabian-s** [comment]
Thanks!

**Comment by Fil** [comment]
This gives me more "feature requests", for example a variable pointSize that would depend on the zoom factor (like deepScatter does, I think?). But out of scope of this issue :)

**Comment by flekschas** [comment]
Thanks for the review!

Since this is my first JOSS review, apologies in advance as I might not know about the best way to address the review.

I got started and put together [a PR](https://github.com/flekschas/regl-scatterplot/pull/108) to address several issues by:
- Fixing types
- Rephrasing that "D3 relies on SVG"
- Adding community guidelines

Should I go ahead and merge the PR or await your feedback first?

> the [observable example](https://observablehq.com/@flekschas/regl-scatterplot) could be enhanced by adding interaction on selection

I agree! I'll get to that tomorrow. Thanks for the quick examples!

> This gives me more "feature requests", for example a variable pointSize that would depend on the zoom factor (like deepScatter does, I think?). But out of scope for this issue :)

Absolutely! It'd be fun to allow customizing the behavior. The good news is that the point size is already dependent on the zoom level. Just not linearly.
_tl/dr:_ Normally, the point size is either zoom invariant (points are always drawn with a radius in absolute pixels) or bound to the geometric zoom. I found both to not be ideal. With zoom invariant point sizes the scatter gets too sparse upon zooming in. On the other hand, having the point size be bound linearly to the geometric zoom means that zooming in won't reveal finer details for points that superimpose each other as they keep superimposing each other at every zoom level. As a solution, in `regl-scatterplot` if the camera scaling is above `1` (the user zooms in) the point size is scaled by the inverse hyperbolic sine function. If the camera scaling is below `0` the point size is linearly scaled according to the camera scaling.

**Comment by Fil** [comment]
Thanks! With this I've checked the last item on my review.

**Comment by fabian-s** [comment]
Wonderful, thank you both for your quick and constructive action!
